src/Simple/JSON.purs

```diff
@@ -96,3 +103,33 @@ instance readFieldsNil ::
   ) => ReadForeignFields Nil row where
   getFields _ _ _ =
     pure $ to {}
+
+
+-- | The purpose of the `ReadKey` class is to turn strings into map keys,
+-- | so we can read generic `Map`s from `StrMap`s.
+class ReadKey a where
+  readKey :: String -> F a
```

Comment:
Could you explain why this needs its own class rather than to use the normal foreign read from purescript-foreign-generic?